Sources/GenericAccessProfile.swift

```diff
@@ -539,6 +909,1740 @@ extension GAPCompleteLocalName: ExpressibleByStringLiteral {
+            if index >= GAPServiceData128BitUUID.uuidLength {
+                serviceData.append(element)
+            }
+        }
+        
+        self.init(uuid: UUID(uuid), serviceData: serviceData)
+    }
+    
+    public var data: Data {
+        let data = UInt128(uuid: uuid).littleEndian.data
+        
+        return serviceData.reduce(data, { $0.0 + [$0.1] })
+    }
+}
+
+extension GAPServiceData128BitUUID: Equatable {
+    
+    public static func == (lhs: GAPServiceData128BitUUID, rhs: GAPServiceData128BitUUID) -> Bool {
+        
+        return lhs.uuid == rhs.uuid && lhs.serviceData == rhs.serviceData
+    }
+}
+
+extension GAPServiceData128BitUUID: CustomStringConvertible {
+    
+    public var description: String {
+        
+        return uuid.description + serviceData.map { String($0) }.description
+    }
+}
+
+/// The Appearance data type defines the external appearance of the device.
+/// This value shall be the same as the Appearance characteristic, as defined in Vol. 3, Part C, Section 12.2.
+public struct GAPAppearance: GAPData {
+    
+    public static let length = MemoryLayout<UInt16>.size
+    
+    public static let dataType: GAPDataType = .appearance
+    
+    public let appearance: UInt16
+    
+    public init(appearance: UInt16) {
+        
+        self.appearance = appearance
+    }
+    
+    public init?(data: Data) {
+        
+        guard data.count == type(of: self).length
+            else { return nil }
+        
+        let appearance = UInt16(littleEndian: UInt16(bytes: (data[0], data[1])))
+        
+        self.init(appearance: appearance)
+    }
+    
+    public var data: Data {
+        
+        let value = appearance.littleEndian
+        return Data(bytes: [value.bytes.0, value.bytes.1])
+    }
+}
+
+extension GAPAppearance: Equatable {
+    
+    public static func == (lhs: GAPAppearance, rhs: GAPAppearance) -> Bool {
+        
+        return lhs.appearance == rhs.appearance
+    }
+}
+
+extension GAPAppearance: CustomStringConvertible {
+    
+    public var description: String {
+        
+        return appearance.description
+    }
+}
+
+/// The Public Target Address data type defines the address of one or more intended recipients of an advertisement when one or more devices were bonded using a public address.
+/// This data type is intended to be used to avoid a situation where a bonded device unnecessarily responds to an advertisement intended for another bonded device.
+///
+/// Size: Multiples of 6 octets
+/// The format of each 6 octet address is the same as the Public Device Address defined in Vol. 6, Part B, Section 1.3.
+///
+/// The public device address is divided into the following two fields:
+/// company_assigned field is contained in the 24 least significant bits
+/// company_id field is contained in the 24 most significant bits
+public struct GAPPublicTargetAddress: GAPData {
+    
+    public static let length = MemoryLayout<UInt8>.size * 6
```

Comment:
Specify with documentation comment that this is the minimum length.

Reply:
@colemancda this is actually the address length (each GAPPublicTargetAddress can contain multiple addresses), so to avoid confusions I'm changing the name to addressLength